infrastructure/foreman/foreman.py

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# (c) 2016, Eric D Helms <ericdhelms@gmail.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: foreman
+short_description: Manage Foreman Resources
+description:
+    - Allows the management of Foreman resources inside your Foreman server.
+version_added: "2.2"
+author: "Eric D Helms (@ehelms)"
+requirements:
+    - nailgun
+    - "python >= 2.7"
+options:
+    server_url:
+        description:
+            - URL of Foreman server.
+        required: true
+    username:
+        description:
+            - Username on Foreman server.
+        required: true
+    password:
+        description:
+            - Password for user accessing Foreman server.
+        required: true
+    entity:
+        description:
+            - The Foreman resource that the action will be performed on (e.g. organization, host).
+        required: true
+    params:
+        description:
+            - Parameters associated to the entity resource to set or edit in dictionary format (e.g. name, description).
+        required: true
+    verify_ssl:
+        description:
+            - When using HTTPS if SSL certificates need to be verified.
+        required: false
+        default: true
+'''
+
+EXAMPLES = '''
+- name: "Create CI Organization"
+  local_action:
+      module: foreman
+      username: "admin"
+      password: "admin"
+      server_url: "https://fakeserver.com"
+      entity: "organization"
+      params:
+        name: "My Cool New Organization"
+'''
+
+RETURN = '''# '''
+
+try:
+    import datetime
+    from nailgun import entities, entity_fields
+    from nailgun.config import ServerConfig
+    HAS_NAILGUN_PACKAGE = True
+except:
+    HAS_NAILGUN_PACKAGE = False
+
+class NailGun(object):
+    def __init__(self, server, entities, module):
+        self._server = server
+        self._entities = entities
+        self._module = module
+
+    def find_organization(self, name, **params):
+        org = self._entities.Organization(self._server, name=name, **params)
+        response = org.search(set(), {'search': 'name={}'.format(name)})
+
+        if len(response) == 1:
+            return response[0]
+        else:
+            self._module.fail_json(msg="No Content View found for %s" % name)
```

Comment:
Oh actually this should just say "No organization found", however, I removed this helper method for now as this behavior doesn't need to be used presently. Thus I updated it to also fix the issue you found below to create an org.